drakrun/drakrun/lib/bindings/xen.py

```diff
@@ -0,0 +1,126 @@
+import logging
+import subprocess
+from typing import Optional
+
+log = logging.getLogger(__name__)
+
+
+def xen_is_vm_running(vm_name: str) -> bool:
+    result = subprocess.run(
+        ["xl", "list", vm_name], capture_output=True
+    )
+    if result.returncode == 0:
+        return True
+    elif b"is an invalid domain identifier" in result.stderr:
+        return False
+    else:
+        raise RuntimeError(f"Unexpected xl list output: {result.stderr}")
+
+
+def xen_create_vm(
+    vm_name: str,
+    config_path: str,
+    pause: bool = False,
+    timeout: Optional[float] = None,
+) -> None:
+    args = [
+        "xl", "create",
+        *(["-p"] if pause else []),
+        config_path
+    ]
+    try:
+        subprocess.run(args, check=True, timeout=timeout)
+    except subprocess.CalledProcessError:
+        raise RuntimeError(f"Failed to launch VM {vm_name}")
+    except subprocess.TimeoutExpired:
+        raise RuntimeError(f"Failed to launch VM {vm_name} within {timeout} seconds")
+
+
+def xen_unpause_vm(vm_name: str, timeout: Optional[float] = None) -> None:
+    try:
+        subprocess.run(["xl", "unpause", vm_name], check=True, timeout=timeout)
+    except subprocess.CalledProcessError:
+        raise RuntimeError(f"Failed to unpause VM {vm_name}")
+    except subprocess.TimeoutExpired:
+        raise RuntimeError(f"Failed to launch VM {vm_name} within {timeout} seconds")
+
+
+def xen_restore_vm(
+    vm_name: str,
+    config_path: str,
+    snapshot_path: str,
+    pause: bool = False,
+) -> None:
+    args = [
+        "xl", "restore",
+        *(["-p"] if pause else []),
+        config_path,
+        snapshot_path
+    ]
+    try:
+        subprocess.run(args, check=True)
```

Comment:
args inline? (like in other places)
`run(X, check=True) == subprocess.check_call(X)` i think